src/parallel.cpp

```diff
@@ -17,38 +17,48 @@
 
 #include <boost/thread/thread.hpp>
 
-#define MAX_SCRIPT_CHECK_QUEUES 4
-
-static CCheckQueue<CScriptCheck> scriptcheckqueue1(128);
-static CCheckQueue<CScriptCheck> scriptcheckqueue2(128);
-static CCheckQueue<CScriptCheck> scriptcheckqueue3(128);
-static CCheckQueue<CScriptCheck> scriptcheckqueue4(128);
-
 using namespace std;
 
+static const unsigned int nScriptCheckQueues = 4;
+std::unique_ptr<CParallelValidation> PV;
 
-void AddScriptCheckThreads(int i, CCheckQueue<CScriptCheck> *pqueue)
+static void HandleBlockMessageThread(CNode *pfrom, const std::string &strCommand, const CBlock &block, const CInv &inv);
+
+static void AddScriptCheckThreads(int i, CCheckQueue<CScriptCheck> *pqueue)
 {
     ostringstream tName;
     tName << "bitcoin-scriptchk" << i;
     RenameThread(tName.str().c_str());
     pqueue->Thread();
 }
 
-void AddAllScriptCheckQueuesAndThreads(int nScriptCheckThreads, boost::thread_group *threadGroup)
+CParallelValidation::CParallelValidation(int threadCount, boost::thread_group *threadGroup)
 {
-    CCheckQueue<CScriptCheck> *pqueue[MAX_SCRIPT_CHECK_QUEUES] = {
-        &scriptcheckqueue1, &scriptcheckqueue2, &scriptcheckqueue3, &scriptcheckqueue4};
+    if (threadCount <= 1)
+        threadCount = 0;
```

Comment:
oh you moved it here...i see..i ididn't notice...it would have better i think to not squash your commits on this one so i could see what you did...there are quite a few changes here.